gui/src/settingsdialog.cpp

```diff
@@ -149,6 +151,21 @@ SettingsDialog::SettingsDialog(Settings *settings, QWidget *parent) : QDialog(pa
 	audio_buffer_size_edit->setPlaceholderText(tr("Default (%1)").arg(settings->GetAudioBufferSizeDefault()));
 	connect(audio_buffer_size_edit, &QLineEdit::textEdited, this, &SettingsDialog::AudioBufferSizeEdited);
 
+	audio_device_combo_box = new QComboBox(this);
+	QList<QString> audio_out_device_list = {};
+	for (QAudioDeviceInfo di : QAudioDeviceInfo::availableDevices(QAudio::AudioOutput))
+	{
+		audio_out_device_list.append(di.deviceName());
+	}	
+	audio_device_combo_box->addItems(audio_out_device_list);
+	auto current_audio_out_device = settings->GetAudioOutDevice();
+	int retrieved_device_index = 0;
+	if(audio_out_device_list.indexOf(current_audio_out_device.deviceName()) >= 0)
+		retrieved_device_index = audio_out_device_list.indexOf(current_audio_out_device.deviceName());
+	audio_device_combo_box->setCurrentIndex(retrieved_device_index);
+	connect(audio_device_combo_box, SIGNAL(currentIndexChanged(int)), this, SLOT(AudioOutputSelected()));
```

Comment:
Use modern connect syntax, remove `AudioOutputSelected()` and use lambda instead
```suggestion
	connect(audio_device_combo_box, &QComboBox::currentIndexChanged, this, [this, audio_device_combo_box]{
		settings->SetAudioOutDevice(audio_device_combo_box->currentData());
	});
```

Reply:
hmm...this change is giving error

```
/home/pi/chiaki/chiaki/gui/src/settingsdialog.cpp:160:80: error: capture of non-variable ‘SettingsDialog::audio_device_combo_box’
  connect(audio_device_combo_box, &QComboBox::currentIndexChanged, this, [this, audio_device_combo_box]{
```

Reply:
Remove `, audio_device_combo_box` from the `[...]` and it should work

Reply:
```
	connect(audio_device_combo_box, &QComboBox::currentIndexChanged, this, [this]{
		settings->SetAudioOutDevice(audio_device_combo_box->currentData());
	});
```
does not work.  Neither does [=]

Reply:
does it need a () with something in it straight after the []   ?

Reply:
```
connect(audio_device_combo_box, &QComboBox::currentIndexChanged, this, [=](){
		settings->SetAudioOutDevice((QString)audio_device_combo_box->currentData().toString());
	});
```
maybe this got me further but with error:
```
/home/pi/chiaki/chiaki/gui/src/settingsdialog.cpp: In constructor ‘SettingsDialog::SettingsDialog(Settings*, QWidget*)’:
/home/pi/chiaki/chiaki/gui/src/settingsdialog.cpp:162:3: error: no matching function for call to ‘SettingsDialog::connect(QComboBox*&, <unresolved overloaded function type>, SettingsDialog*, SettingsDialog::SettingsDialog(Settings*, QWidget*)::<lambda()>)’
  })
```

Reply:
This seems a lot harder than it should be. Why is there an overload error when its just a qt native class?
I read other people have issues with that too. (unless I'm doing it wrong)

Reply:
Yes you are right, coding blind is hard. Fixed now.